sdk/go/dagger/querybuilder/marshal.go

```diff
@@ -0,0 +1,59 @@
+package querybuilder
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+)
+
+func MarshalGQL(v any) string {
```

Comment:
Awesome -- I just checked and I think it would work.

We could use it in `func (s *Selection) Build()` to construct the query.

`MarshalGQL` would still exist and be pretty similar to today (AFAIK gqlparser doesn't provide any Go->AST helper), but it would construct an `ast.Value` rather than a string.

Should've used it from the get go -- the reason it's this way is because I started with the `graphb` library, later hit a wall, and re-implemented parts of it which were using string construction.

Since we're a bit short in time, I filed #3332 so we can get this merged and get people using it, will clean up later.